File: app/controllers/tools/ToolState.java
package controllers.tools;

import model.user.Role;

public enum ToolState {

  // @formatter:off
  LIVE("Verfügbare Tools", Role.USER),
  ALPHA("Tools in Alpha-Status", Role.ADMIN),
  BETA("Tools in Beta-Status",Role.ADMIN),
  EXPERIMENTAL("Experimentelle Tools", Role.SUPERADMIN);
  // @formatter:on

  private String description;
  private Role requiredRole;

  private ToolState(String theDescription, Role theRequiredRole) {
    description = theDescription;
    requiredRole = theRequiredRole;
  }

  public String getDescription() {
    return description;
  }

  public Role getRequiredRole() {
    return requiredRole;
  }

}

File: server/app/model/toolMains/ToolState.scala
package model.toolMains

import enumeratum.{EnumEntry, PlayEnum}
import model.Role

sealed abstract class ToolState(val german: String, val greek: String, requiredRole: Role) extends EnumEntry

object ToolState extends PlayEnum[ToolState] {

  override val values: IndexedSeq[ToolState] = findValues


  case object LIVE extends ToolState("Verfügbare Tools", "", Role.RoleUser)

  case object ALPHA extends ToolState("Tools in Alpha-Status", "&alpha;", Role.RoleAdmin)

  case object BETA extends ToolState("Tools in Beta-Status", "&beta;", Role.RoleAdmin)

}
